com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java

```diff
@@ -200,10 +208,73 @@ public void acceptSearchMatch(SearchMatch match) {
         return resolved.toArray(new String[resolved.size()]);
     }
 
-    private static boolean isRuntime(final IClasspathEntry classpathEntry) {
+    /**
+     * There is an issue on isTest: it will return true if the scope is runtime, so we will this method for testing whether
+     * the classpath entry is for test only.
+     *
+     * @param classpathEntry classpath entry
+     * @return whether this classpath entry is only used in test
+     */
+    private static boolean forTestOnly(final IClasspathEntry classpathEntry) {
         for (IClasspathAttribute attribute : classpathEntry.getExtraAttributes()) {
-            if (SCOPE_ATTRIBUTE.equals(attribute.getName()) && "runtime".equals(attribute.getValue())) {
-                return true;
+            if (GRADLE_SCOPE_ATTRIBUTE.equals(attribute.getName()) || MAVEN_SCOPE_ATTRIBUTE.equals(attribute.getName())) {
+                return TEST_SCOPE.equals(attribute.getValue());
+            }
+        }
+        return classpathEntry.isTest();
+    }
+
+    /**
+     * Test whether the main class is located in test folders.
+     * @param project the java project containing the main class
+     * @param mainClass the main class name
+     * @return whether the main class is located in test folders
+     */
+    private static boolean isMainClassInTestFolder(IJavaProject project, String mainClass) {
+        // get a list of test folders and check whether main class is here
+        int constraints = IJavaSearchScope.SOURCES;
+        List<IJavaElement> testFolders = new ArrayList<IJavaElement>();
+        try {
+            IPackageFragmentRoot[] packageFragmentRoot = project.getPackageFragmentRoots();
+            for (int i = 0; i < packageFragmentRoot.length; i++) {
+                if (packageFragmentRoot[i].getElementType() == IJavaElement.PACKAGE_FRAGMENT_ROOT
+                        && packageFragmentRoot[i].getKind() == IPackageFragmentRoot.K_SOURCE) {
+                    IClasspathEntry cpe = packageFragmentRoot[i].getResolvedClasspathEntry();
+                    if (forTestOnly(cpe)) {
+                        testFolders.add(packageFragmentRoot[i]);
+                    }
+                }
+            }
+        } catch (JavaModelException e) {
+            // ignore
+        }
+        if (!testFolders.isEmpty()) {
+            try {
+
+                List<Object> mainClassesInTestFolder = new ArrayList<>();
```

Comment:
boolean isFound = false;

Reply:
I cannot change the value isFound at the callback